### PR TITLE
Added stopwatch

### PR DESCRIPTION
**Pull request by Panky-codes**
Addresses issue: #144 
Added a stopwatch with all the necessary functions such as play, pause, lap, stop. 

For now, I hijacked the position of the paddle app. Let me know where the stopwatch belong in the app list :)

### CODE REVIEW

**Comment by JF002** [comment]
Looks awesome! Thanks for this PR and sorry for the delay. I'll try to review it soon ;)

File: src/displayapp/screens/ApplicationList.cpp [hunk 1]
```diff
@@ -60,7 +60,8 @@ std::unique_ptr<Screen> ApplicationList::CreateScreen2() {
            {Symbols::asterisk, Apps::Meter},
            {Symbols::paintbrush, Apps::Paint},
                   {Symbols::info, Apps::Notifications},
-                  {Symbols::paddle, Apps::Paddle},
+                  //TODO: Need to find the right place based on comments from JF
```
Comment (JF002) [inline src/displayapp/screens/ApplicationList.cpp:63]: You can take the slot of the 'Meter' app. This app is a simple demo app I added as a placeholder when there was no other apps available in InfiniTime. Now, we can remove it :)

That way, we'll keep the game Paddle in place ;)

File: src/displayapp/screens/StopWatch.h [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+#include "../LittleVgl.h"
+
+#include "FreeRTOS.h"
+#include "portmacro_cmsis.h"
+
+#include <array>
+
+namespace Pinetime::Applications::Screens {
+
+  enum class States { INIT, RUNNING, HALTED };
```
Comment (JF002) [inline src/displayapp/screens/StopWatch.h:14]: Other enum classes do not name their values with CAPITAL letters. Could you rename these as 'Init', 'Running' and 'Halted' for more coherence?
Same for Events.

File: src/displayapp/screens/StopWatch.h [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+          // This transformation is to ensure that head is always pointing to index 0.
+          const auto transformed_idx = (head - idx) % capacity;
+          return (&_arr[transformed_idx]);
+        }
+      }
+      return nullptr;
+    }
+
+  private:
+    std::array<TimeSeparated_t, N> _arr;
```
Comment (JF002) [inline src/displayapp/screens/StopWatch.h:58]: Also for more consistency with the rest of the code : could you rename `_arr` as `array' for example? And maybe also `currentSz` to `currentSize`?

**Comment by JF002** [comment]
I've just tested and reviewed your PR : it works as expected, and the code is pretty straightforward.
For the position on the menu, you can replace the Meter app by this new Stopwatch.
Not a big deal, but I've also added some comments about the naming of the variables.

Thanks!

**Comment by Panky-codes** [comment]
Thanks for your comments. I made all the changes and also tested it once. 

Just a note, I did a git pull of your develop branch and now I also see the lvgl submodules have been updated (src/libs/lvgl) in my PR. I didn't expect them but could you verify if it is ok or if I messed something up when I pulled your changes?

**Comment by JF002** [comment]
Thanks for the fixes!
I'm not sure I understand what is going on with the submodule and why it was updated in your PR. I'll try to understand that before merging and potentially breaking something...

**Comment by Panky-codes** [comment]
Hi JF, 
   Apparently, git pull doesn't really update the submodule. I went into the submodule and updated it to point to the same commit as your branch. And now there is no diff anymore for lvgl libs. 

I think now it is ready to be merged.  :)

File: src/displayapp/DisplayApp.cpp [hunk 4]
```diff
@@ -206,7 +207,8 @@ void DisplayApp::RunningState() {
       case Apps::Meter: currentScreen.reset(new Screens::Meter(this)); break;
       case Apps::Twos: currentScreen.reset(new Screens::Twos(this)); break;
       case Apps::Paint: currentScreen.reset(new Screens::InfiniPaint(this, lvgl)); break;
-      case Apps::Paddle: currentScreen.reset(new Screens::Paddle(this, lvgl)); break;
+      //TODO: Change it back
```
Comment (Panky-codes) [inline src/displayapp/DisplayApp.cpp:210]: I just realized I forgot to revert this. I Will do it now.

**Comment by JF002** [comment]
> Apparently, git pull doesn't really update the submodule. I went into the submodule and updated it to point to the same commit as your branch. And now there is no diff anymore for lvgl libs.

I think you have to do a `git module update` after the pull to ensure that your submodule is up to date with the one from the repo.
There is no changes for the submodule in your PR so I guess that's good now :)

**Comment by Panky-codes** [comment]
I did do the git submodule update, but somehow nothing changed. Anyway, I think everything is in order now and I also tested the paddle app so that I didn't introduce any regression. 

I think now the PR is definitely ready to be merged :)

**Pull request merged**